File: Lesson4/looper/src/main/java/ru/mirea/volkovmm/looper/MainActivity.java
package ru.mirea.volkovmm.looper;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import ru.mirea.volkovmm.looper.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private MyLooper myLooper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());


        Handler mainThreadHandler = new Handler(Looper.getMainLooper()) {
            @Override
            public void handleMessage(Message msg) {
                String result = msg.getData().getString("result");
                Log.d(MainActivity.class.getSimpleName(), "Результат из MyLooper: " + result);


                Toast.makeText(MainActivity.this, result, Toast.LENGTH_LONG).show();
            }
        };


        myLooper = new MyLooper(mainThreadHandler);
        myLooper.start();


        binding.buttonMirea.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String ageStr = binding.editTextAge.getText().toString();
                String profStr = binding.editTextProfession.getText().toString();

                if (ageStr.isEmpty() || profStr.isEmpty()) return;

                int age = Integer.parseInt(ageStr);


                Message msg = Message.obtain();
                Bundle bundle = new Bundle();
                bundle.putInt("AGE", age);
                bundle.putString("PROFESSION", profStr);
                msg.setData(bundle);


                if (myLooper != null && myLooper.mHandler != null) {
                    myLooper.mHandler.sendMessage(msg);
                    Toast.makeText(MainActivity.this, "Задача отправлена в фон!", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Lesson2/toastapp/src/main/java/ru/mirea/volkovmm/toastapp/MainActivity.kt
package ru.mirea.volkovmm.toastapp

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText = findViewById<EditText>(R.id.editText)
        val btnCount = findViewById<Button>(R.id.btnCount)

        btnCount.setOnClickListener {
            val textLength = editText.text.length

            val resultMessage = "СТУДЕНТ № 11 ГРУППА БСБО-50-24 Количество символов - $textLength"

            Toast.makeText(this, resultMessage, Toast.LENGTH_SHORT).show()
        }
    }
}